src/minecraft/kotlin/xyz/wagyourtail/unimined/internal/minecraft/patch/reindev/FoxLoaderMinecraftTransformer.kt

```diff
@@ -0,0 +1,179 @@
+
+    override var loadingPlugin: String = ""
+
+    private val jarTask = project.tasks.maybeCreate("jar".withSourceSet(provider.sourceSet), Jar::class.java)
+
+    override fun transform(minecraft: MinecraftJar): MinecraftJar {
+        val transformed = super.transform(minecraft)
+
+        val name = transformed.path.fileName.toString().substringBeforeLast('.')
+        val newPath = Path(transformed.path.toString().replace(name, "$name+FoxLoaderPatched"))
+
+        project.logger.info(transformed.path.absolutePathString())
+        project.logger.info(newPath.absolutePathString())
+
+        project.javaexec {
+            it.classpath = foxLoaderInvoker
+            it.mainClass.set("com/fox2code/foxloader/invoker/Main")
+            it.args(
+                transformed.path.absolutePathString(),
+                newPath.absolutePathString(),
+                "false", isClient.toString()
+            )
+        }
+
+        return transformed.copy(path = newPath)
+    }
+
+    override fun apply() {
+        jarTask.manifest {
+            project.logger.lifecycle("Applying attributes")
+            it.attributes["For-FoxLoader-Version"] = foxLoader.dependencies.first().version
+            it.attributes["For-ReIndev-Version"] = provider.version
+            if (modId.isEmpty()) throw IllegalStateException("Mod ID must be set for source set: ${provider.sourceSet.name}")
+            it.attributes["ModId"] = this.modId
+            if (commonMod.isNotEmpty()) it.attributes["CommonMod"] = this.commonMod
+            if (isClient && clientMod.isNotEmpty()) it.attributes["ClientMod"] = this.clientMod
+            if (isServer && serverMod.isNotEmpty()) it.attributes["ServerMod"] = this.serverMod
+            if (modName.isNotEmpty()) it.attributes["ModName"] = this.modName
+            if (modVersion.isNotEmpty()) it.attributes["ModVersion"] = this.modVersion
+            if (modDescription.isNotEmpty()) it.attributes["ModDesc"] = this.modDescription
+            if (modWebsite.isNotEmpty()) it.attributes["ModWebsite"] = this.modWebsite
+            if (preClassTransformer.isNotEmpty()) it.attributes["PreClassTransformer"] = this.preClassTransformer
+            if (loadingPlugin.isNotEmpty()) it.attributes["LoadingPlugin"] = this.loadingPlugin
+            if (unofficial) it.attributes["Unofficial"] = "true"
+        }
+
+        super.apply()
+    }
+
+    override fun applyClientRunTransform(config: RunConfig) {
+        if (!canCombine) {
+            val task = project.tasks.getByName("runClient".withSourceSet(provider.sourceSet))
+            task.dependsOn(jarTask)
+        } else {
+            TODO("FoxLoader 2.0 will hopefully not rely on the jar task")
+        }
+        super.applyClientRunTransform(config)
+        config.mainClass.set(mainClass)
+        config.jvmArgs(
+            "-Dfoxloader.dev-mode=true",
+            "-Dfoxloader.inject-mod=${jarTask.outputs.files.singleFile.absolutePath}"
```

Comment:
the problem with this is, there's no way to depend on/inject dependency mods in gradle

Reply:
Hum... I guess FoxLoader never supported that in it's dev plugin, weird I thought I added this feature.

I'll fix that for `1.3.0`, and make something for `FoxLoaderInvoker` to backport this to older versions.